lib/middleware/operation.js

```diff
@@ -32,55 +36,92 @@ function findPropertyOperations ({ types, method, term, resource }) {
   return operations
 }
 
-function factory (api) {
-  return async (req, res, next) => {
-    if (!req.hydra.resource) {
-      return next()
+function mapOperations ({ api, res, method }) {
+  return res.locals.hydra.resources.flatMap((resource) => {
+    let moreOperations
+    if ('property' in resource) {
+      moreOperations = findPropertyOperations({ resource, method, api }).toArray()
+    } else {
+      const types = clownface({ ...api, term: [...resource.types] })
+      moreOperations = findClassOperations(types, method).toArray()
+    }
+
+    if (moreOperations.length) {
+      return moreOperations.map(operation => ({ resource, operation }))
     }
 
+    return [{ resource, operation: null }]
```

Comment:
Keeps the resources without any operations around to later determine if they are subject or object usage